ksqldb-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -303,21 +303,25 @@ private void ensureKeySchemasMatch(
         .getSchemaTranslator(keyFormat.getFormatInfo().getProperties())
         .toParsedSchema(keySchema);
 
+    final Optional<SchemaMetadata> latest;
     try {
-      final SchemaMetadata latest = serviceContext.getSchemaRegistryClient()
-          .getLatestSchemaMetadata(
-              KsqlConstants.getSRSubject(dataSource.getKafkaTopicName(), true));
-      if (!latest.getSchema().equals(schema.canonicalString())) {
-        throw new KsqlException("Cannot INSERT VALUES into data source " + dataSource.getName()
-            + ". ksqlDB generated schema would overwrite existing key schema."
-            + "\n\tExisting Schema: " + latest.getSchema()
-            + "\n\tksqlDB Generated: " + schema.canonicalString());
-      }
-    } catch (IOException | RestClientException e) {
+      latest = SchemaRegistryUtil.getLatestSchema(
+          serviceContext.getSchemaRegistryClient(),
+          dataSource.getKafkaTopicName(),
+          true);
+
+    } catch (final Exception e) {
       maybeThrowSchemaRegistryAuthError(format, dataSource.getKafkaTopicName(), true, "read", e);
```

Comment:
What's the purpose of separately handling potential auth errors here rather than unifying in SchemaRegistryUtil where missing schemas are handled?

Reply:
It makes it a little easier to handle the error message. If it's an auth error, I want to expose that directly. If it's another error, then I want to throw our generic shim: "Could not determine that insert values..."

If I handled this in the SchemaRegistryUtil, then I would have to throw a different exception type or parse the exception message to decide whether to propagate it or to wrap it.